components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.spec.ts

```diff
@@ -35,18 +40,164 @@ describe('CreateChefServerModalComponent', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('chef.internal');
+
+        createForm.controls['ip_address'].setValue('1.2234.3.4');
+        errors = createForm.controls['ip_address'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['pattern']).toBeTruthy();
+      });
+
+      // Many testing ideas attributed to https://mathiasbynens.be/demo/url-regex
+
+      using([
+        ['is using something other than http or https', 'httpld://www.chef.io'],
+        ['contains two periods', 'chef..internal'],
+        ['there is no TLD suffix', 'http://foo.com.'],
+        ['contains hypens in the TLD', 'chef.this-will-not'],
+        ['has a TLD that is longer than 25 characters', 'chef.thisisareallylongtldandwontwork'],
+        ['has a TLD that is shorter than 2 characters', 'chef.i'],
+        ['has numbers in the TLD', 'chef.017'],
+        ['has a port number that is too high', 'https://chef.io:987274892'],
+        ['has a colon but no port number', 'https://chef.io:'],
+        ['has a letter in the port', 'https://chef.io:123a'],
+        ['has no domain', 'http://'],
+        ['has no secure domain', 'https://'],
+        ['domain is dots', 'https://..'],
+        ['domain is hash', 'http://#'],
+        ['domain has a space', 'http:// shouldfail.net'],
+        ['contains all numbers', 'http://10.1.1.0']
+      ], function (description: string, input: string) {
+        it(('when the fqdn ' + description), () => {
+          createForm.controls['name'].setValue('test');
+          createForm.controls['id'].setValue('test');
+          createForm.controls['ip_address'].setValue('1.2.3.4');
+
+          createForm.controls['fqdn'].setValue(input);
+          errors = createForm.controls['fqdn'].errors || {};
+
+          expect(createForm.valid).toBeFalsy();
+          expect(errors['pattern']).toBeTruthy();
+        });
+      });
+    });
+
+
+
+    describe('the form should be valid', () => {
+      it('when all 4 inputs are filled and valid', () => {
+        expect(createForm.valid).toBeFalsy();
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('chef.internal');
+        createForm.controls['ip_address'].setValue('1.2.3.4');
+        expect(createForm.valid).toBeTruthy();
+      });
+
+      using([
+        ['has a TLD that is longer than 2 and less than 25 characters', 'chef.internal'],
+        ['uses https', 'https://chef.io'],
+        ['uses http', 'http://chef.io'],
+        ['omits http or https', 'chef.thisworks'],
+        ['has a port number', 'https://chef.io:123'],
+        ['contains hyphens in the domain', 'new-company-who-dis.nice'],
+        ['contains underscores in the domain', 'new_company_who_dis.chef'],
+        ['contains digits in the domain', '8675309.jenny'],
```

Comment:
😮 `8675309.jenny` is valid (this line) but `chef.017` (L137) is not; curious...

Reply:
As it sits today, this is by our design.  

Currently, the TLD cannot be all numbers, so `chef.017` will fail.  
Conversely, a TLD of all letters should pass as long as it is between 2 and 25 characters.

Reply:
looking at the Regex though, we certainly could update the TLD portion
from
`(https?:\/\/)?[a-zA-Z0-9_]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z]{2,25}(:[0-9]{1,5})?(\/.*)?$`

to something like
`(https?:\/\/)?[a-zA-Z0-9_]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z` change here -> `0-9` <-  `]{2,25}(:[0-9]{1,5})?(\/.*)?$`